src/lxc/start.c

```diff
@@ -561,11 +558,7 @@ struct lxc_handler *lxc_init_handler(const char *name, struct lxc_conf *conf,
 	for (i = 0; i < LXC_NS_MAX; i++)
 		handler->nsfd[i] = -1;
 
-	handler->name = strdup(name);
-	if (!handler->name) {
-		ERROR("failed to allocate memory");
-		goto on_error;
-	}
+	handler->name = name;
```

Comment:
It makes me uneasy, since the container* is shared among threads in a way the handler isn't, and the c->name was previously dereferenced under container_mem_lock().  I *think* it's fine - or rather I think if it isn't we have other problems - but I'm not convinced.  We can try it and just keep this in mind if weird failures end up happening.